services/guard/guard.go

```diff
@@ -0,0 +1,93 @@
+package guard
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"github.com/dedis/cothority/sda"
+	"github.com/dedis/crypto/abstract"
+)
+
+// This file contains all the code to run a Guard service. The Guard receives takes a
+// request for the Guard password protection service and sends back the message received encrypted with the key for each service
+// in the server.
+
+// ServiceName is the name to refer to the Guard service.
+const ServiceName = "Guard"
+
+func init() {
+	sda.RegisterNewService(ServiceName, newGuardService)
+	network.RegisterMessageType(&Request{})
+	network.RegisterMessageType(&Response{})
+}
+
+//This is the area where Z is generated for a server, it creates z, which is a bytestring of length n for each guard.
+
+// Guard is a structure that stores the guards secret key, z, to be used later in the process of hashing the clients requests
+type Guard struct {
+	*sda.ServiceProcessor
+	path string
+	z    []byte
+}
+
+// Request is what the Guard service is expected to receive from clients.
+type Request struct {
+	UID   []byte
+	Epoch []byte
+	Msg   []byte
+}
+
+// Response is what the Guard service will reply to clients.
+type Response struct {
+	Msg []byte
+}
+
+//[]byte("gmPTvzEuu88GAL_67UJFFZXzaAhWn_N3ZonPU-D272Y=SuNSfNfzkP3CSSNIR3uuERNxDWzN7MwebioZGkZBwdc=")
+
+// Request treats external request to this service.
+func (st *Guard) Request(e *network.ServerIdentity, req *Request) (network.Body, error) {
+	//hashy computes the hash that should be sent back to the main server H(pwhash, x, UID, Epoch)
+	hashy := abstract.Sum(network.Suite, req.Msg, st.z, req.UID, req.Epoch)
+	key := []byte("example key 1234")
```

Comment:
You'll have to explain once again why you need this key here. I don't see it used anywhere.